compiler/modules/front/codeGen.bal

```diff
@@ -103,9 +103,11 @@ class CodeGenContext {
         self.returnType = returnType;
     }
 
-    function createRegister(bir:SemType t, string? varName = ()) returns bir:Register {
+    function createRegister(bir:SemType t, bir:PositionString? varName = ()) returns bir:Register {
         bir:Register reg = bir:createRegister(self.code, t, varName);
-        self.registerVarNames[reg.number] = varName;
+        if varName != () {
+            self.registerVarNames[reg.number] = varName[1];
```

Comment:
But you're accessing this array in validLoopAssignments, which needs the position.

Do we need the PositionString in the registerVarNames array or in the bir:Register or both?

Reply:
added a new function `registerPosition` to get the register position (leaving `registerVarName` to return the register name). If we have a reference to the `bir:Register` having  PositionString there is enought. But in cases (such as `validLoopAssignments` that only has the register number we need something else to get the position).